components/match2/commons/match_group_input.lua

```diff
@@ -241,6 +241,38 @@ function MatchGroupInput.getInexactDate(suggestedDate)
 	return getContentLanguage():formatDate('c', inexactDateString)
 end
 
+--[[
+Parses the match group context. The match group context describes where a match
+group is relative to the tournament page.
+]]
+function MatchGroupInput.readContext(args)
+	local context = {
+		bracketIndex = tonumber(globalVars:get('match2bracketindex')) or 0,
+		groupRoundIndex = MatchGroupInput.readGroupRoundIndex(args),
+		matchSection = args.matchsection or globalVars:get('matchsection'),
+		sectionHeader = args.section or globalVars:get('bracket_header'),
+		tournamentParent = globalVars:get('tournament_parent'),
+	}
+
+	globalVars:set('bracket_header', context.sectionHeader)
+	globalVars:set('matchsection', context.matchSection)
```

Comment:
Doesn't really make sense to have `readContext` set vars

Reply:
Good point. I'll move it to `MatchGroupInput.saveContext`